### PR TITLE
Issue #5 Goto URL with multiple query string parameters incorrectly decoding

### PR DESCRIPTION
**Pull request by OGIS-RyoKobayashi**
## Analysis
 If redirect by Goto URL with multiple query , there has a problem what missing part of queries.
 Goto ( goto / gotoOnFail ) URL be supposed to be encoded.
 Similarly, this occurs with SAML federatied occurring in "SSORedirect" or "SSOPOST".

## Solution
-  OPENAM-9597 Fixing SSO redirect
-  OPENAM-9597 Goto URL with multiple query string parameters incorrectly decoded
-  Change to use Redux

## Testing
Log in using multiple query string goto parameters and ensure that they are redirected with the query parameters remaining. 

In addition, there is no difference in the URL before and after the end of the session when redirecting including the specified parameters.

```
<before login>
http://openam.example.net/openam/XUI/#login/&goto=http%3A%2F%2Fhogehoge.example.com%3Fhoge%3D1%26fuga%3D2
```
```
<after>
http://hogehoge.example.com?hoge=1&fuga=2
```

### CODE REVIEW

**Comment by tsujiguchitky** [comment]
Please add the copyright to the license header of each file.

File: openam-ui/openam-ui-ria/src/main/js/org/forgerock/openam/ui/user/login/RESTLoginView.js [hunk 1]
```diff
@@ -295,13 +295,11 @@ define([
...
-                routeToLoginUnavailable(
-                    RESTLoginHelper.filterUrlParams (
-                        URIUtils.parseQueryString (
-                            URIUtils.getCurrentCompositeQueryString()
-                        )
-                    )
-                );
+                var queryParam = URIUtils.parseQueryString(URIUtils.getCurrentCompositeQueryString());
+                if (queryParam.goto) {
+                    queryParam.goto = encodeURIComponent(queryParam.goto);
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/js/org/forgerock/openam/ui/user/login/RESTLoginView.js]: Why don't you encode parameters except goto?

File: openam-ui/openam-ui-ria/src/main/js/org/forgerock/openam/ui/user/login/RESTLoginView.js [hunk 2]
```diff
@@ -417,6 +415,10 @@ define([
         handleUrlParams () {
             var urlParams = URIUtils.parseQueryString(URIUtils.getCurrentCompositeQueryString());
 
+            if (urlParams.goto) {
+                urlParams.goto = encodeURIComponent(urlParams.goto);
```
Comment (tsujiguchitky) [inline openam-ui/openam-ui-ria/src/main/js/org/forgerock/openam/ui/user/login/RESTLoginView.js]: This fix seems to break the process for SSORedirect.

see https://github.com/openam-jp/openam/commit/13422162b919a611421e551ab7e050e0a0a11376

**Comment by OGIS-RyoKobayashi** [comment]
Thanks reviewing!
I changed the scope of patch for OPENAM-9597.
Removed change / add commit related to redux implementation.

**Comment by tsujiguchitky** [comment]
Looks good to me.

**Comment by tsujiguchitky** [comment]
#5

**Pull request merged**